modin/engines/base/frame/partition_manager.py

```diff
@@ -18,12 +18,44 @@
 
 from modin.error_message import ErrorMessage
 from modin.data_management.utils import compute_chunksize
-from modin.config import NPartitions, ProgressBar
+from modin.config import NPartitions, ProgressBar, BenchmarkMode
 
 from pandas.api.types import union_categoricals
 import os
 
 
+def wait_computations_if_benchmark_mode(func):
+    """
+    Synchronize performing a function in benchmark mode.
+
+    Parameters
+    ----------
+    func: callable
+        A function that should be performed in syncronous mode.
+
+    Note
+    ----
+        `func` should return numpy array with partitions.
+    """
+    if BenchmarkMode.get():
+
+        def wait(*args, **kwargs):
+            """Wait for computation results."""
+            result = func(*args, **kwargs)
+            if isinstance(result, tuple):
```

Comment:
Why is this check necessary?

Reply:
To distinguish situations where one or more values are returned.

Reply:
Which situations do you mean? Please, give an example.

Reply:
In instance, `from_pandas` and `map_partitions`

Reply:
I see, it makes sense for `from_pandas` but not for `map_partitions`. The latter returns np.array.